src/cascadia/TerminalSettingsEditor/MainPage.idl

```diff
@@ -3,10 +3,11 @@
 
 namespace Microsoft.Terminal.Settings.Editor
 {
-    [default_interface]
-    runtimeclass MainPage : Windows.UI.Xaml.Controls.Page
+    [default_interface] runtimeclass MainPage : Windows.UI.Xaml.Controls.Page
     {
         MainPage(Microsoft.Terminal.Settings.Model.CascadiaSettings settings);
         event Windows.Foundation.TypedEventHandler<Object, Microsoft.Terminal.Settings.Model.SettingsTarget> OpenJson;
+
+        static Microsoft.Terminal.Settings.Model.CascadiaSettings Settings { get; };
```

Comment:
I'm not sufficiently pleased with the answers I've gotten to this question on other PRs

**why is this static**

who owns it

who makes sure it doesn't point to an old Settings object after SUI is closed

this is not what you're supposed to be doing

this is _global state_

Reply:
So there's two things related to this:
1. cloning `CascadiaSettings`
  - this part is not done yet. Right now, we're hooked up to the `CascadiaSettings` used by TermApp.
  - when `MainPage` gets created in a `SettingsTab`, we _need_ a reference to `TermApp`'s settings object to be able to clone it and replace it when we want to apply the settings.
2. hooking up all of the Settings UI to that clone
  - we need a way for all of the Settings UI to data bind to the clone. We're exposing the clone with a static `Settings` getter.
  - NOTE: _right now_ we're still hooked up to `TermApp`'s settings object. This is temporary. It makes testing way easier right now. And lets us not have to serialize our settings to see changes (which isn't done yet)

I do agree, it's a bit weird that we have a static reference to the clone. I'm not sure how else we can have shared access to it though. :/

Reply:
I mean, there's the tried and true method of passing it into the constructor for every page, or giving it in the navigation arguments...

Reply:
that... sounds like what we should be doing, actually hahaha. You ok with me adding that to the board? It'll definitely have conflicts with all the data binding we're doing, and I'd rather have it be its own PR.

Reply:
Add it to the board.